### PR TITLE
fix several gcc-13 compile issues

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Enterprise companion PR: https://github.com/arangodb/enterprise/pull/1408

Fix several compile issues with gcc-13.
Fix for `LogContext.h` provided by @mpoeter.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: -
  - [ ] Backport for 3.10: -

#### Related Information

- [ ] Docs PR: 
- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/1408
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Cluster/ClusterMethods.cpp [hunk 1]
```diff
@@ -466,7 +466,7 @@ OperationResult handleCRUDShardResponsesSlow(
 
     int nrok = 0;
     auto commError = TRI_ERROR_NO_ERROR;
-    fuerte::StatusCode code;
+    fuerte::StatusCode code = fuerte::StatusUndefined;
```
Comment (jsteemann) [inline arangod/Cluster/ClusterMethods.cpp:469]: Fixes issue about `code` potentially being uninitialized (false positive).

File: arangod/RestServer/LanguageCheckFeature.cpp [hunk 2]
```diff
@@ -155,10 +156,11 @@ std::tuple<std::string, LanguageType> getOrSetPreviousLanguage(
     arangodb::ArangodServer& server, std::string_view collatorLang,
     LanguageType currLangType) {
   std::string prevLanguage;
-  LanguageType prevType;
+  LanguageType prevType = LanguageType::INVALID;
```
Comment (jsteemann) [inline arangod/RestServer/LanguageCheckFeature.cpp:159]: Fixes issue about `prevType` potentially being uninitialized (false positive).

File: lib/Logger/LogContext.h [hunk 3]
```diff
@@ -240,7 +240,7 @@ struct LogContext::OverloadVisitor : GenericVisitor<OverloadVisitor<Overloads>>,
   explicit OverloadVisitor(Overloads overloads)
       : GenericVisitor<OverloadVisitor>(), Overloads(std::move(overloads)) {}
   template<class T>
-  void visit(std::string_view const& key, T&& value) const {
+  void doVisit(std::string_view const& key, T&& value) const {
```
Comment (jsteemann) [inline lib/Logger/LogContext.h:243]: Fixes issue about hidden virtual member from base class.

**Review by neunhoef — APPROVED** [review]
LGTM

**Pull request merged**